File: c/12/twelve.c
#include <iostream>
#include <ostream>


int triangle(int n)
{
  int sum = 0;
  for(int i = 0; i <= n; i++) 
  {
    sum += i;
  }
  return sum;
}

int divisorsfn(int n)
{
  int numdivs = 0;
  int i = 0;
  if (n == 1){
    return 1;
  }
  while (true)
  {
    i += 1;
    if (i == n) {
      break;
    }

    if (n % i == 0){
      numdivs++;
    }
  }
  return numdivs;
}


int main() 
{
  int i = 0;
  while(true) {
    i++;
    int numsum = triangle(i);
    int divs = divisorsfn(numsum);
    std::cout << numsum  << " " << divs << std::endl;
    if (divs >= 500)
    {
      std::cout << numsum << std::endl;
      break;
    }
  }
}

File: python/12/twelve.py


def triangle(n):
  return sum(xrange(n+1))

def divisorsfn(n, dup = False):
  divisors = []
  i = 0
  if n == 1:
    return {1: 1}
  while 1:
    i += 1
    if dup:
      break
    if i == n:
      break
    if n % i == 0:
      divisors[i:i+1] = [i]
  return divisors


i = 0
while True:

  i+=1

  numsum = triangle(i)

  divs = divisorsfn(numsum)
  size = len(divs)
  print numsum, size
  if size >= 500:
    print numsum
    break
